### PR TITLE
Composite algo

### PR DESCRIPTION
**Pull request by ZEXINLIU**
The predictor-corrector algorithm (PCA) is ready for merging into master.

### CODE REVIEW

File: UncertainSCI/utils/compute_mom.py [hunk 1]
```diff
@@ -0,0 +1,63 @@
+import numpy as np
+import scipy.special as sp
+
+from UncertainSCI.utils.compute_subintervals import compute_subintervals
+from UncertainSCI.utils.quad import gq_modification_composite
+from UncertainSCI.utils.quad import gq_modification_unbounded_composite
+
+def compute_mom_bounded(a, b, weight, k, singularity_list, Nquad=10):
```
Comment (dcwhite) [inline UncertainSCI/utils/compute_mom.py:8]: Are these functions for computing moments? (As mentioned in one of the doc comments). If so, I'd prefer the function name spelled that out, the goofy abbreviation "mom" is not worth the saved 3-4 characters.

File: UncertainSCI/utils/compute_subintervals.py [hunk 2]
```diff
@@ -0,0 +1,72 @@
+import numpy as np
+
+def compute_subintervals(a, b, singularity_list):
```
Comment (dcwhite) [inline UncertainSCI/utils/compute_subintervals.py:3]: See, this is a better function name.

File: UncertainSCI/utils/compute_subintervals.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    strength_left  = np.array([entry[1] for entry in singularity_list])
+    strength_right = np.array([entry[2] for entry in singularity_list])
+
+    # We can discard any singularities that lie to the left of a or the right of b
+    discard = []
+    for (ind,s) in enumerate(singularities):
+        if s < a-tol or s > b+tol:
+            discard.append(ind)
+
+    singularities  = np.delete(singularities, discard)
```
Comment (dcwhite) [inline UncertainSCI/utils/compute_subintervals.py:22]: This code could use a refactoring--the same multi-step procedure is done multiple times to 3 different arrays. That smells like a couple methods/classes needing to be extracted.

**Review by dcwhite — COMMENTED** [review]
Minor code commentary; I'll let @akilnarayan decide if required for merging.

**Comment by dcwhite** [comment]
Also, a couple test failures and warnings to fix:
```______________________ ERROR collecting tests/test_ttr.py ______________________
ImportError while importing test module '/home/runner/work/UncertainSCI/UncertainSCI/tests/test_ttr.py'.
Hint: make sure your test modules/packages have valid Python names.
Traceback:
/opt/hostedtoolcache/Python/3.6.12/x64/lib/python3.6/importlib/__init__.py:126: in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
tests/test_ttr.py:8: in <module>
    from UncertainSCI.mthd_mod_correct import compute_ttr_bounded, compute_ttr_unbounded
E   ModuleNotFoundError: No module named 'UncertainSCI.mthd_mod_correct'
=============================== warnings summary ===============================
UncertainSCI/utils/quad.py:38
  /home/runner/work/UncertainSCI/UncertainSCI/UncertainSCI/utils/quad.py:38: DeprecationWarning: invalid escape sequence \i
    """
```

**Comment by dcwhite** [comment]
The new code has a ton of flake/lint errors as well.

File: tests/test_ttr.py [hunk 4]
```diff
@@ -0,0 +1,132 @@
+import unittest
+
+import numpy as np
+
+from UncertainSCI.families import jacobi_weight_normalized, jacobi_recurrence_values
+from UncertainSCI.families import hermite_recurrence_values
+
+from UncertainSCI.mthd_mod_correct import compute_ttr_bounded, compute_ttr_unbounded
```
Comment (akilnarayan) [inline tests/test_ttr.py:8]: Like Dan noticed, this generates an error -- there doesn't appear to be a `mthd_mod_correct` module in `UncertainSCI`, so this throws an error.

File: UncertainSCI/ttr/compute_ttr.py [hunk 5]
```diff
@@ -0,0 +1,672 @@
...
+
+from UncertainSCI.utils.compute_subintervals import compute_subintervals
+from UncertainSCI.utils.quad import gq_modification_composite, \
+        gq_modification_unbounded_composite
+from UncertainSCI.utils.array_unique import array_unique
+
+"""
+Predictor-corrector method
+"""
+def predict_correct_bounded(a, b, weight, N, singularity_list, Nquad=10):
```
Comment (akilnarayan) [inline UncertainSCI/ttr/compute_ttr.py:15]: Are there tests for any of these? I couldn't find anything. Even a `git grep lanczos` only popped up the definitions in this file, but didn't show anything else, i.e., nothing is calling these functions. Can you write some basic tests? Even just sanity checks are fine for now.

File: tests/test_ttr.py (hunk 4 above)
Comment (ZEXINLIU) [reply tests/test_ttr.py:8]: Yes, I could not even remember where this came from. But anyway I delete them.

File: UncertainSCI/ttr/compute_ttr.py (hunk 5 above)
Comment (ZEXINLIU) [reply UncertainSCI/ttr/compute_ttr.py:15]: A quick test for PC and orthogonality was written in test_ttr.py.

**Review by akilnarayan — APPROVED** [review]
Still many flake8 errors but we can fix those later.

**Review by ZEXINLIU — COMMENTED** [review]
where shall we put ttr/compute_ttr.py?

**Comment by akilnarayan** [comment]
Sorry, forgot to add it back it. I made a new pull request you can approve. The new file is in `UncertainSCI/ttr.py`

**Pull request merged**